classes/renderables/topic_render.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * Class topic_render
+ * @package    mod
+ * @subpackage hsuforum
+ * @author    khendricks@2u.com
+ * @copyright Copyright (c) 2U 2u.com
+ */
+
+class topic_render {
+    // Should always follow default button class of css theme.
+    const BUTTON_CLASS = 'btn btn-primary';
+
+    /**
+     * topic_subcription_button
+     * Builds new subscription button html desktop/mobile.
+     * @param string $last_reply_html
+     * @param null $currently_subbed
+     * @return string
+     */
+    public function topic_subcription_button($last_reply_html = '', $currently_subbed = null) {
+
+        $button_group = '';
+        if(!empty($currently_subbed)) {
+            $button_group .= '<div class="d-block d-lg-block d-xl-block subscriber-wrapper">
+                                <div class="last-reply-block">' . $last_reply_html . '</div>
+                                    <button type="button" class="trigger-subscribe rounded-pill ' . self::BUTTON_CLASS . '">'
+                                . get_string('topicfollowing','hsuforum') .
+                                '</button>
+                              </div>
+                              <div class="d-none d-sm-block d-md-none mobile-btn subscriber-wrapper">
+                                 <div class="last-reply-block">' . $last_reply_html . '</div>
+                                    <button  type="button" class="trigger-subscribe rounded-pill ' . self::BUTTON_CLASS . '">'
+                                    . get_string('topicfollowing','hsuforum') .
+                                    '</button>
+                              </div>';
```

Comment:
Is is possible to use templates within this function?

Reply:
I guess it would be possible would you like me to make it a template?